base/src/com/google/idea/blaze/base/sync/projectstructure/ModuleValidator.java

```diff
@@ -0,0 +1,14 @@
+package com.google.idea.blaze.base.sync.projectstructure;
+
+import com.intellij.openapi.extensions.ExtensionPointName;
+import com.intellij.openapi.module.Module;
+
+/**
+ * Extension point to customize the validity of a given module, used to show the module in project tree view.
+ */
+public interface ModuleValidator {
```

Comment:
I don't think `validity` is a proper term here...
I'd go with `ModuleManager` and instead of `isModuleValid` something like `isModuleSupported` or `isManaged`

Reply:
There is already another ModuleManager...
I updated the method to 'hasOtherContentRoot' as thats really what is being checked to tag them as orphan or not.

Reply:
I'm still open for suggestions on class names..

Reply:
You have package/namespace... Yet name is something to select carefully here. EP are generic so let's have something generic yet specific enough.

Reply:
Yeah, so I'd prefer to be as explicit as possible. So how about sth like `isOwnedByExternalPlugin` or sth like that? I agree that using the term "validity" might be misleading

Reply:
I'm thinking something like this:
class name - `ExternalModuleProvider`, method - `shouldOrphanModule` or `isAnOrphanModule` (using word 'orphan' because orphan modules will not be shown in this context)